tools/pylint-extensions/pylint-guidelines-checker/pylint_guidelines_checker.py

```diff
@@ -389,18 +400,25 @@ def visit_asyncfunctiondef(self, node):
         node.decoratornames() returns a set of the method's decorator names.
 
         :param node: function node
-        :type node: ast.FunctionDef
+        :type node: ast.AsyncFunctionDef
         :return: None
         """
         try:
-            if node.parent.name.endswith("Client") and node.is_method() and not node.name.startswith("_") and \
-                    node.parent.name not in self.ignore_clients:
-                if node.args.kwarg and "azure.core.tracing.decorator_async.distributed_trace_async" not in \
-                        node.decoratornames() and "builtins.classmethod" not in node.decoratornames():
-                    self.add_message(
-                        msgid="client-method-missing-tracing-decorator-async", node=node, confidence=None
-                    )
-        except AttributeError:
+            path = node.root().name
+            split_path = path.split(".")
+            new_path = ".".join(split_path[:len(split_path)-1])
+            if new_path.count("_") == 0:
+                if node.parent.name.endswith("Client") and node.is_method() and not node.name.startswith("_") and \
+                        node.parent.name not in self.ignore_clients:
+                    if node.args.kwarg and node.name not in self.ignore_functions and not node.name.endswith("client") \
+                        and self.ignore_decorators[0] not in node.decoratornames() \
+                            and self.ignore_decorators[1] not in node.decoratornames() \
+                                and self.ignore_decorators[3] not in node.decoratornames():
```

Comment:
Same comment here. This should be simplified so it doesn't need to be changed if `ignore_decorators` changes.